llvm/utils/TableGen/DXILEmitter.cpp

```diff
@@ -91,15 +98,12 @@ static ParameterKind getParameterKind(const Record *R) {
     return ParameterKind::I32;
   case MVT::fAny:
   case MVT::iAny:
+  case MVT::Any:
     return ParameterKind::Overload;
-  case MVT::Other:
-    // Handle DXIL-specific overload types
-    if (R->getValueAsInt("isHalfOrFloat") || R->getValueAsInt("isI16OrI32")) {
-      return ParameterKind::Overload;
-    }
-    LLVM_FALLTHROUGH;
   default:
-    llvm_unreachable("Support for specified DXIL Type not yet implemented");
+    report_fatal_error(
+        "Support for specified parameter type not yet implemented",
+        /*gen_crash_diag*/ false);
```

Comment:
This should still be `llvm_unreachable` - if we get here we made a mistake in `DXIL.td` or otherwise broke our invariants.